runtime/mainnet/src/pos/session_payout/mod.rs

```diff
@@ -18,6 +21,37 @@ type PositiveImbalanceOf<T> =
 
 type CurrencyOf<T> = <T as pallet_staking::Config>::Currency;
 
+#[derive(Encode, Decode, Default, Clone, TypeInfo, MaxEncodedLen, PartialEq, Eq, Debug)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+pub enum ValidatorCommissionAlgorithm {
+	Static(Perbill),
+	#[default]
+	Median,
+}
+
+impl ValidatorCommissionAlgorithm {
+	pub fn comission<T: Config>(&self, era: EraIndex) -> Option<Perbill> {
+		match &self {
+			Self::Static(comission) => Some(*comission),
+			Self::Median => {
+				// to work out properly, we need to map all current validators to their actual comission (not the era one)
+				// TODO: Open question. MEDIAN of current ERA or current SESSION? Can validator change preferences during session? I think - yes, cause ERA is long.
+				let current_validators = pallet_staking::ErasStakers::<T>::iter_key_prefix(era);
+				let mut comissions = current_validators
+					.map(|id| pallet_staking::Validators::<T>::get(id).commission)
+					.collect::<Vec<_>>();
+
+				if comissions.is_empty() {
+					return None;
+				}
+				comissions.sort_unstable();
+				let median = comissions[comissions.len() / 2];
```

Comment:
```suggestion
				let median = if comissions.len() % 2 == 1 {
				  comissions[comissions.len() / 2]
				} else {
				  comissions[comissions.len() / 2 - 1].saturating_add(comissions[comissions.len() / 2 + 1]) / 2
				}
```

Reply:
See [wiki](https://en.wikipedia.org/wiki/Median) for details.

Reply:
Done